osm2gtfs/core/configuration.py

```diff
@@ -31,6 +33,64 @@ def __init__(self, args):
         # Validate and prepare start and end date
         self._prepare_dates()
 
+        # Initiate variable for schedule source information
+        self._schedule_source_file = None
+
+    def get_schedule_source(self, refresh=False):
+        """Loads the schedule source information.
+
+        Loads a schedule source file from either a path or a url specified
+        in the config file
+
+        :return schedule_source_file: The schedule source file object.
+
+        """
+
+        if 'schedule_source' not in self.config:
+            sys.stderr.write("Error: No schedule source specified in config file.\n")
+            sys.exit(0)
+
+        else:
+            # Omit schedule_source import if indicated in config file
+            if self.config['schedule_source'] == "None":
+                return None
+
+            # Preferably return cached data about routes
+            if refresh is False:
+                # Check if _schedule_source_file data is already built in this object
+                if not self._schedule_source_file:
+                    # If not, try to get _schedule_source_file from file cache
+                    self._schedule_source_file = Cache.read_file(
+                        'schedule-source-' + self.config['selector'])
+                # Return cached data if found
+                if bool(self._schedule_source_file):
+                    return self._schedule_source_file
+
+            # No cached data was found or refresh was forced
+            print("Load schedule source information from " +
+                  self.config['schedule_source'])
+
+            # Check if local file exists
+            if os.path.isfile(self.config['schedule_source']):
+
+                # Open file and add to config object
+                with open(self.config['schedule_source'], 'r') as schedule_source_file:
+                    self._schedule_source_file = schedule_source_file
```

Comment:
Does this leak the file object outside the `with` environment and leave it open?

Reply:
Yes, is there any better way to do it?

Reply:
I am not sure it is a good idea to pass file objects around this way. Why not passing its content around instead?

Reply:
Because, we don't know at this point the format of the schedule file. It can be json, it can be csv or any other format, which then will be used in the (overridden) `schedule_creator`

Reply:
Good point. In this case, maybe just use a string and leave it to the creator to interpret it. So it can handle file opening and closing itself.

Reply:
I went for reading the files already at this point, but not decoding it. This is necessary to do proper caching and it avoids passing around any file objects. Hopefully this is also fine?!

Reply:
Yes, thanks!